exp/api/v1alpha3/azuremanagedmachinepool_webhook.go

```diff
@@ -77,76 +71,14 @@ func (r *AzureManagedMachinePool) ValidateUpdate(oldRaw runtime.Object, client c
 					"field is immutable"))
 		}
 	}
-
-	if r.Spec.Mode != string(NodePoolModeSystem) && old.Spec.Mode == string(NodePoolModeSystem) {
-		// validate for last system node pool
-		if err := r.validateLastSystemNodePool(client); err != nil {
-			allErrs = append(allErrs, field.Invalid(
-				field.NewPath("Spec", "Mode"),
-				r.Spec.Mode,
-				"Last system node pool cannot be mutated to user node pool"))
-		}
-	}
-
 	if len(allErrs) != 0 {
 		return apierrors.NewInvalid(GroupVersion.WithKind("AzureManagedMachinePool").GroupKind(), r.Name, allErrs)
 	}
-
 	return nil
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
 func (r *AzureManagedMachinePool) ValidateDelete(client client.Client) error {
 	azuremanagedmachinepoollog.Info("validate delete", "name", r.Name)
-
-	if r.Spec.Mode != string(NodePoolModeSystem) {
-		return nil
-	}
-
-	return errors.Wrapf(r.validateLastSystemNodePool(client), "if the delete is triggered via owner MachinePool please refer to trouble shooting section in https://capz.sigs.k8s.io/topics/managedcluster.html")
-}
-
-// validateLastSystemNodePool is used to check if the existing system node pool is the last system node pool.
-// If it is a last system node pool it cannot be deleted or mutated to user node pool as AKS expects min 1 system node pool.
-func (r *AzureManagedMachinePool) validateLastSystemNodePool(cli client.Client) error {
-	ctx := context.Background()
-
-	// Fetch the Cluster.
-	clusterName, ok := r.Labels[clusterv1.ClusterLabelName]
-	if !ok {
-		return nil
-	}
-
-	ownerCluster := &clusterv1.Cluster{}
-	key := client.ObjectKey{
-		Namespace: r.Namespace,
-		Name:      clusterName,
-	}
-
-	if err := cli.Get(ctx, key, ownerCluster); err != nil {
-		if azure.ResourceNotFound(err) {
-			return nil
-		}
-		return err
-	}
-
-	if !ownerCluster.DeletionTimestamp.IsZero() {
```

Comment:
So this check deletionTimestamp is not enough? we need to remove the whole check? I just want to keep our fork as much consistent with upstream as possible.